x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx

```diff
@@ -38,96 +36,63 @@ export const getDefaultQuery = (): CspFindingsRequest & { groupBy: GroupBy } =>
     }),
   },
   {
-    value: 'resourceType',
-    label: i18n.translate('xpack.csp.findings.groupBySelector.groupByResourceTypeLabel', {
-      defaultMessage: 'Resource Type',
+    value: 'resource',
+    label: i18n.translate('xpack.csp.findings.groupBySelector.groupByResourceIdLabel', {
+      defaultMessage: 'Resource ID',
     }),
   },
 ];
 
-const getFindingsBaseEsQuery = ({
-  query,
-  dataView,
-  filters,
-  queryService,
-}: Pick<CspFindingsRequest, 'filters' | 'query'> & {
-  dataView: DataView;
-  queryService: CspClientPluginStartDeps['data']['query'];
-}) => {
-  if (query) queryService.queryString.setQuery(query);
-  queryService.filterManager.setFilters(filters);
-
-  try {
-    return {
-      index: dataView.title,
-      query: buildEsQuery(
-        dataView,
-        queryService.queryString.getQuery(),
-        queryService.filterManager.getFilters()
-      ),
-    };
-  } catch (error) {
-    return {
-      error:
-        error instanceof Error
-          ? error
-          : new Error(
-              i18n.translate('xpack.csp.findings.unknownError', {
-                defaultMessage: 'Unknown Error',
-              })
-            ),
-    };
-  }
-};
-
 export const FindingsContainer = ({ dataView }: { dataView: DataView }) => {
   const { euiTheme } = useEuiTheme();
   const groupByOptions = useMemo(getGroupByOptions, []);
-  const {
-    data,
-    notifications: { toasts },
-  } = useKibana().services;
-
-  const {
-    urlQuery: { groupBy, ...findingsQuery },
-    setUrlQuery,
-  } = useUrlQuery(getDefaultQuery);
+  const { urlQuery, setUrlQuery } = useUrlQuery(getDefaultQuery);
 
-  const baseQuery = useMemo(
-    () => getFindingsBaseEsQuery({ ...findingsQuery, dataView, queryService: data.query }),
-    [data.query, dataView, findingsQuery]
+  const baseEsQuery = useMemo(
+    () => ({
+      index: dataView.title,
+      // TODO: this will throw for malformed query
+      // page will display an error boundary with the JS error
+      // will be accounted for before releasing the feature
+      query: buildEsQuery(dataView, urlQuery.query, urlQuery.filters),
+    }),
+    [dataView, urlQuery]
   );
 
-  const countResult = useFindingsCounter(baseQuery);
-  const findingsResult = useFindings({
-    ...baseQuery,
-    size: findingsQuery.size,
-    from: findingsQuery.from,
-    sort: findingsQuery.sort,
+  const findingsGroupByResource = useFindingsByResource({
+    ...baseEsQuery,
+    enabled: urlQuery.groupBy === 'resource',
   });
 
-  useEffect(() => {
-    if (baseQuery.error) {
-      toasts.addError(baseQuery.error, { title: TEXT.SEARCH_FAILED });
-    }
-  }, [baseQuery.error, toasts]);
+  const findingsCount = useFindingsCounter({
+    ...baseEsQuery,
+    enabled: urlQuery.groupBy === 'none',
+  });
+
+  const findingsGroupByNone = useFindings({
+    ...baseEsQuery,
+    enabled: urlQuery.groupBy === 'none',
+    size: urlQuery.size,
+    from: urlQuery.from,
+    sort: urlQuery.sort,
+  });
```

Comment:
`useFindingsCounter` may be removed when we add the distribution bar to the `resource` table by plugging the agg query directly in `useFindings` or `useFindingsCounter`